source/Ship.cpp

```diff
@@ -4969,3 +4969,14 @@ double Ship::CalculateDeterrence() const
 		}
 	return tempDeterrence;
 }
+
+
+
+bool Ship::Immitates(const Ship &other) const
+{
+	for(auto &it : attributes.Attributes())
+		if(other.Attributes().Get(it.first) != it.second)
+			return false;
+
+	return true;
+}
```

Comment:
Might it make more sense to match on model name and outfit counts rather than attributes? With this, you could theoretically have two ships of different models that have a combination of outfits that cause them to match one another on all their attributes. The attributes of a ship also don't take into account different weaponry.

Reply:
Yes, that might be both simpler and more accurate. I will rework this patch.